zipkin4net/Criteo.Profiling.Tracing/ServerTrace.cs

```diff
@@ -29,4 +31,43 @@ public void Dispose()
             Trace.Record(Annotations.ServerSend());
         }
     }
+
+    public static class ServerTraceExtensions
+    {
+        /// <summary>
+        /// Runs the action and adds an error annotation in case of failure
+        /// </summary>
+        /// <param name="serverTrace"></param>
+        /// <param name="action"></param>
+        public static void TracedAction(this ServerTrace serverTrace, Action action)
```

Comment:
Not sure we'll need the synchronised version. Can you add the usage of these method in this PR ?